apigentools/cli.py

```diff
@@ -72,10 +82,33 @@ def cli(ctx, **kwargs):
     ctx.obj = dict(kwargs)
     toplog = logging.getLogger(__name__.split(".")[0])
     set_log(toplog)
+    check_min_version(ctx)
     if ctx.obj.get("verbose"):
         set_log_level(toplog, logging.DEBUG)
 
 
+def check_min_version(click_ctx):
+    with change_cwd(click_ctx.obj.get("spec_repo_dir")):
+        configfile = os.path.join(
+            os.path.join(constants.SPEC_REPO_CONFIG_DIR, constants.DEFAULT_CONFIG_FILE)
+        )
+        try:
+            config = Config.from_file(configfile)
+        except OSError:
+            check_for_legacy_config(click_ctx, configfile)
+
+        # Version like - "apigentools, version 0.10.1.dev27+dirty"
+        min_version = config.raw_dict.get("apigentools_min_version", "0.0.0")
```

Comment:
```suggestion
        min_version = config.raw_dict.get("version", "0.0.0")
```

Reply:
Hmm, I'm not sure `version` is specific enough since it's going to live as a field in the spec config file. I.e. version could imply its the version of the config file itself, not the tooling that uses it. 
Definitely open to improving this name though.

Reply:
what is the difference between tool version and config version in our case?

Reply:
Talked offline. I think it's mostly up to the spec owner to define how loose or strict they want to be with how the strict the minimum version is of the tooling. But since indeed the tooling is tightly combined with the spec config file, I changed to `version`